app/main.py

```diff
@@ -1,44 +1,65 @@
+import asyncio
 import time
+from typing import Awaitable, Any
 
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
-    # create an IOT service
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function # Coroutine obj -> not callable!
```

Comment:
You're trying to await a coroutine object, but you're not calling it. The `function` in this context is a coroutine object, not a coroutine function. You should call the coroutine function to get a coroutine object that you can await. Add parentheses after `function` to call it.